addon/components/flexberry-layerslegends.js

```diff
@@ -34,5 +34,44 @@ export default Ember.Component.extend({
     @type Object[]
     @default null
   */
-  layers: null
+  layers: null,
+
+  /**
+    Visible not group layers.
+
+    @property visibleLayers
+    @type Object[]
+  */
+  visibleLayers: Ember.computed('layers.[]', function() {
```

Comment:
may be passing map or maplayers object and filter it be better than recursive search on tree?